teamcity-rest-client-api/src/main/kotlin/org/jetbrains/teamcity/rest/api.kt

```diff
@@ -50,6 +52,10 @@ interface VcsRootLocator {
 interface BuildLocator {
     fun fromConfiguration(buildConfigurationId: BuildConfigurationId): BuildLocator
 
+    fun withNumber(buildNumber: String): BuildLocator
```

Comment:
Why do you need this method in `BuildLocator`? We already have `TeamCityInstance::build(buildType, number)` method which will return a build by its number.

Reply:
You may also get multiple builds per build number, they're not unique

Reply:
I know. If it's a real case then ok.

Reply:
From my point of view, `withNumber` is easier to discover especially when one wants to have more constraints like a since date and so on.